### ISSUE TITLE
Crashes when marking season watched

### ISSUE THREAD

**Issue by jlafitte** [jellyfin/jellyfin-androidtv#2409]
### Describe the bug

1. Go to a TV show season
2. Mark as watched
3. Crashes
4. Go back in, and it looks like it did mark as watched even though it crashed.

### Logs

[Log.txt](https://github.com/jellyfin/jellyfin-androidtv/files/10398328/Log.txt)

### Application version

0.15.2

### Where did you install the app from?

Google Play

### Device information

Nvidia shield 2017

### Android version

11

### Jellyfin server version

10.8.8

**Comment by nielsvanvelzen** [comment]
The crash logs show that the server is not responsive when retrieving item info, not something to do with marking as watched.

**Comment by jlafitte** [comment]
> The crash logs show that the server is not responsive when retrieving item info, not something to do with marking as watched.

It only happens when doing it for a season with a large amount of episodes.  I think it's just taking a while to confirm the action completed.  I don't know what fix would be, allowing a longer wait or just letting it go in the background?

**Comment by nielsvanvelzen** [comment]
The timeout is currently set at 30 seconds, so I'd say this is a backend issue because a simple "mark as watched" request shouldn't take that long. Dealing with the timeout is something we should catch client side though.

**Comment by jlafitte** [comment]
Yeah, it does take a little time even for one episodes (2-3 seconds maybe).  I'm using the docker image and haven't really made any changes.  I'll take a look around and see if I can work out that issue.

**Comment by systemcrash** [comment]
Also had the same

**Comment by nielsvanvelzen** [comment]
Just to be sure; does this only happens when marking as watched via the long-press menu or also when using the button beneath the episode/show/movie description?

**Comment by systemcrash** [comment]
For me it's the long press. Then k-boom

**Comment by jlafitte** [comment]
Short answer: Only long press on seasons causes the crash, after about 10 seconds.

tl;dr answer:
For series, I only seem to be able to do it with a button.  When I do, it doesn't update immediately, but it doesn't crash.  Eventually if I wait long enough, the button will change to red and show it's watched.

For seasons, it crashes after long press. I only seem to be able to it from long press, I don't get a button.  Not sure if it used to or not, I might have changed a setting.  If I go into a season I just see an episode listing.

I can mark individual episodes watched either long press or the button.  It does kind of "lock up" when I say mark as played from long press, but it comes back within 2-3 seconds, no crash.  With the button, it doesn't update immediately but it doesn't seem to lock up the interface, I can move around and the button updates to red in 2-3 seconds.

For the record, from the web interface, it is marks them watched/unwatched immediately, and both the context and button works fine.

I really haven't dug into the code, but there seems to be a difference between what happens on long press menu option and the button.

It was mentioned earlier that there was a 30 second timeout, but the crash happens at about 10 seconds when I tested.

**Comment by nielsvanvelzen** [comment]
It looks like the timeout is actually set to 10 seconds right now, it used 30 seconds before. I've made a change that changes it back to those 30 seconds. It's not a perfect solution but should hopefully prevent some of the crashes for now.

There's some more changes that will need to happen client side to properly deal with timeouts (like not crashing, showing a message to the user etc.)

**Comment by TheMalone** [comment]
This is still an issue on a Sheield Pro. Will it ever be fixed?
